evap/grades/templates/grades_course_view.html

```diff
@@ -29,7 +29,11 @@ <h3 class="mb-3">{{ course.name }} ({{ semester.name }})</h3>
                                 <td>
                                     <a href="{% url 'grades:download_grades' grade_document.id %}" class="btn btn-sm btn-light" data-bs-toggle="tooltip" data-bs-placement="top" title="{% trans 'Download' %}"><span class="fas fa-download"></span></a>
                                     {% if user.is_grade_publisher %}
-                                        <a href="{% url 'grades:edit_grades' semester.id course.id grade_document.id %}" class="btn btn-sm btn-light" data-bs-toggle="tooltip" data-bs-placement="top" title="{% trans 'Edit' %}"><span class="fas fa-pencil-alt"></span></a>
+                                        {% if final_grades %}
+                                            <a href="{% url 'grades:edit_grades' semester.id course.id grade_document.id %}?final=true" class="btn btn-sm btn-light" data-bs-toggle="tooltip" data-bs-placement="top" title="{% trans 'Edit' %}"><span class="fas fa-pencil-alt"></span></a>
+                                        {% else %}
+                                            <a href="{% url 'grades:edit_grades' semester.id course.id grade_document.id %}" class="btn btn-sm btn-light" data-bs-toggle="tooltip" data-bs-placement="top" title="{% trans 'Edit' %}"><span class="fas fa-pencil-alt"></span></a>
```

Comment:
These two lines are almost the same, you could also inline the `if` into the url:
```django
href="{% url 'grades:edit_grades' semester.id course.id grade_document.id %}?final={% if final_grades %}true{% else %}false{% end %}"
```

Maybe you could even just render variable into a string using `{{ final_grades }}`. On that note: The variable should probably be called `is_final_grades` or similar to suggest it is a boolean and not a collection of the actual grades.